### PR TITLE
Support for "bucket_sort" and "bucket_selector" pipeline aggregations

### PR DESCRIPTION
**Pull request by matthinea**
bucket_sort docs: https://www.elastic.co/guide/en/elasticsearch/reference/7.12/search-aggregations-pipeline-bucket-sort-aggregation.html
bucket_selector docs: https://www.elastic.co/guide/en/elasticsearch/reference/7.12/search-aggregations-pipeline-bucket-selector-aggregation.html

### CODE REVIEW

**Comment by matthinea** [comment]
@dcromer lemme know what you think.

File: test/arelastic/aggregations/bucket_sort_test.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'helper'
+
+class Arelastic::Aggregations::BucketSortTest < Minitest::Test
+  def test_as_elastic
+    with_hash_sort = Arelastic::Aggregations::BucketSort.new('foo_agg', 'size' => 3, 'from' => 5, 'sort' => {
+      'bar_bucket' => 'desc', 'baz_bucket' => 'asc', 'fiz_bucket' => { 'order' => 'asc' }
+    })
```
Comment (dcromer) [inline test/arelastic/aggregations/bucket_sort_test.rb:7]: what is this whitespace lol

File: test/arelastic/aggregations/bucket_sort_test.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            { 'baz_bucket' => 'desc' },
+            { 'fiz_bucket' => {'order' => 'desc'} }
+          ]
+
+        }
+      }
+    }
+    assert_equal expected, with_array_sort.as_elastic
+  end
+end
```
Comment (dcromer) [inline test/arelastic/aggregations/bucket_sort_test.rb:40]: If you just need to test parse_sort, write a smaller test(s) for that.  I think the current tests are a bit overkill.

**Review by dcromer — APPROVED** [review]
Looks good, I see where you're going with this.

**Comment by matthinea** [comment]
Cool. I was going to just leave it at that. I can't merge this though, @ebarendt can maybe or @matthuhiggins.

File: lib/arelastic/aggregations/bucket_sort.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+      def as_elastic_aggregation
+        bucket_sort = {}
+        bucket_sort['sort'] = parse_sort if sort
+        bucket_sort['from'] = from if from
+        bucket_sort['size'] = size if size
+        { 'bucket_sort' => bucket_sort }
+      end
+
+      def parse_sort
```
Comment (ebarendt) [inline lib/arelastic/aggregations/bucket_sort.rb:22]: Should this be private? I see a test for it below, but that test should be part of the public interface?

Comment (matthinea) [reply lib/arelastic/aggregations/bucket_sort.rb]: Made it private.

File: test/arelastic/aggregations/bucket_sort_test.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+          'from' => 5,
+          'size' => 3
+        }
+      }
+    }
+    assert_equal expected, aggregation.as_elastic
+  end
+
+  def parse_sort(sort)
+    Arelastic::Aggregations::BucketSort.new('foosort', sort: sort).send(:parse_sort)
```
Comment (ebarendt) [inline test/arelastic/aggregations/bucket_sort_test.rb:23]: Hmm.. I was thinking more of testing the behavior by sending the data through the public method.

Comment (matthinea) [reply test/arelastic/aggregations/bucket_sort_test.rb]: Ah gotcha.

File: lib/arelastic/aggregations/bucket_sort.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          if sort.is_a?(Hash)
+            sort.map { |bucket_path, order| { bucket_path => order } }
+          elsif sort.is_a?(String)
+            [sort]
+          elsif sort.is_a?(Array)
+            sort
+          else
+            raise TypeError.new('Expecting a hash, string, or array as the "sort" argument')
+          end
+        end
```
Comment (matthuhiggins) [inline lib/arelastic/aggregations/bucket_sort.rb:34]: Simply support the same type of params that a normal search sort does.

**Review by matthuhiggins — COMMENTED** [review]
This broke some conventions with the project. See https://github.com/matthuhiggins/arelastic/commit/f848856cdbadb56af1cba51abc754930714def59, which I accidentally pushed to master when meaning to make a PR.

Let me know what you think of my follow-up changes.

If the project has conventions that should change, I can help make a wholesale change. For example, I've wanted to use keyword arguments in more places. The way symbol parameters worked  before Ruby 3 was too much of a pain to get right.

File: lib/arelastic/aggregations/bucket_selector.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+module Arelastic
+  module Aggregations
+    class BucketSelector < Arelastic::Aggregations::Aggregation
+      attr_accessor :script_params, :script
```
[the remark this answers is not in the document]
Comment (matthinea) [reply lib/arelastic/aggregations/bucket_selector.rb:4]: No, they can be removed.

**Pull request merged**